File: CF/D_Balanced_Round.py
for _ in range(int(input())):
    n,k = map(int,input().split(" "))

    a = list(map(int,input().split(" ")))

    bad = []
    #bad.append(-1)

    

    a.sort()

    flag = True

    for i in range(n-1):
        if a[i+1] - a[i] > k:
            bad.append(i)
            flag = False

    #print(a)
    

   #print(bad)

    if flag:
        print(0)
        continue



    ans = 1e9

    #print(a)
    #print(bad)
   
    
    for i in range(len(bad)-1):
        ans = min(ans,n-(bad[i+1]-bad[i]))
        #print(bad[i] + 1 + (n - bad[i+1]))
    

    
    if(len(bad) > 1):
        ans = min(ans,bad[-1]+1,n-bad[0]-1)
    
    
    elif(len(bad) == 1):
        ans = min(ans,n-bad[0]-1,bad[0]+1)
    print(ans)

File: CF/D_Balanced_Round.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

int main(){
    int tt; cin >> tt;

    while(tt--){
        int n, k; cin >> n >> k;

        vector<int> a(n);

        for(int i = 0; i < n;i ++){
            cin >> a[i];
        }

        if(n == 1){
            cout << "0\n";
            continue;
        }

        sort(a.begin(),a.end());

        vector<int> bad;

        for(int i = 0; i < n-1; i++ ){
            if(a[i+1] - a[i] > k){
                bad.push_back(i);
            }
        }

        int ans = 1e9;

        cout << "good\n";

        
         for(int i = 0; i < bad.size()-1; i++){
            cout << bad[i] << endl;
        }
        

       

        cout << ans << "\n";

        
    }
}